http/obmc_shell.hpp

```diff
@@ -0,0 +1,242 @@
+#pragma once
+
+#include <pty.h>
+#include <pwd.h>
+#include <termios.h>
+
+#include <app.hpp>
+#include <boost/asio/posix/stream_descriptor.hpp>
+#include <boost/process/io.hpp>
+#include <websocket.hpp>
+
+#include <map>
+
+namespace crow
+{
+
+namespace obmc_shell
+{
+
+class Handler : public std::enable_shared_from_this<Handler>
+{
+  public:
+    Handler(crow::websocket::Connection* conn) :
+        session(conn), streamFileDescriptor(conn->getIoContext()),
+        doingWrite(false)
+    {}
+
+    ~Handler() = default;
+
+    void doClose()
+    {
+        streamFileDescriptor.close();
+
+        // boost::process::child::terminate uses SIGKILL, need to send SIGTERM
+        int rc = kill(pid, SIGKILL);
+        session = nullptr;
+        if (rc)
+        {
+            return;
+        }
+        waitpid(pid, nullptr, 0);
+    }
+
+    void connect()
+    {
+        pid = forkpty(&ttyFileDescriptor, nullptr, nullptr, nullptr);
+
+        if (pid == -1)
+        {
+            // ERROR
+            if (session)
+            {
+                session->close("Error creating ssh child process");
+            }
+            return;
+        }
+        if (pid == 0)
+        {
+            // CHILD
+
+            // sets the effective user ID
+            // as service user
+            bool isUidSet = false;
+            if (auto userName = session->getUserName(); !userName.empty())
+            {
+                if (struct passwd* pw = getpwnam(userName.c_str());
+                    pw != nullptr)
+                {
+                    int uidr = setuid(pw->pw_uid);
+                    if (!uidr)
+                    {
+                        isUidSet = true;
+                    }
+                    else
+                    {
+                        BMCWEB_LOG_ERROR << "sets service user failed Error: "
```

Comment:
Don't you want to return in these error cases and not keep going?

Reply:
isUidSet will not set true so it will retun after the closing session 
` if (!isUidSet)
            {
                session->close("sets service user failed");
                return;
            }
`